pkg/kubelet/kubelet.go

```diff
@@ -1957,8 +1962,14 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	// Use WithoutCancel instead of a new context.TODO() to propagate trace context
 	// Call the container runtime's SyncPod callback
 	sctx := context.WithoutCancel(ctx)
+	syncStartTime := kl.clock.Now()
```

Comment:
we only need this with the feature gate. Should it be hidden under the feature gate as well? Also - what is the logic of starting time here and not any earlier?